File: src/api/news-survey-send.ts
import type { NewsSurvey, SessionHandle } from "~/models";
import { newsRemoteMutate } from "./private/news-remote-mutate";

/**
 * Answers the survey with the given answers.
 * By default, it'll answer with the questions that were given when the survey was created.
 *
 * You can either manipulate the questions directly or pass in your own answers.
 */
export const newsSurveySend = async (session: SessionHandle, survey: NewsSurvey, alsoMarkAsRead = true): Promise<void> => {
  await newsRemoteMutate(session, survey, {
    markAsRead: alsoMarkAsRead,
    onlyMarkAsRead: false,
    delete: false
  });

  if (alsoMarkAsRead)
    survey.read = true;
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/news-survey-send.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.api.private.NewsRemoteMutateOptions
import ink.literate.pawnote.api.private.newsRemoteMutate
import ink.literate.pawnote.models.NewsItem
import ink.literate.pawnote.models.SessionHandle

/**
 * Answers the survey with the given answers. By default, it'll answer with the questions that were
 * given when the survey was created.
 *
 * You can either manipulate the questions directly or pass in your own answers.
 */
suspend fun newsSurveySend(
    session: SessionHandle,
    survey: NewsItem<Any>,
    alsoMarkAsRead: Boolean = true
) {
  newsRemoteMutate(
      session,
      survey,
      NewsRemoteMutateOptions(markAsRead = alsoMarkAsRead, onlyMarkAsRead = false, delete = false))

  if (alsoMarkAsRead) survey.read = true
}
